### PR TITLE
Update person version on person updates and send value to Kafka

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

We don't currently do anything with this version value in CH (because the schema change hasn't been merged), but we should start producing the messages with it before merging the change.

## How did you test this code?

*Please describe.*

### CODE REVIEW

**Comment by yakkomajuri** [comment]
The failing tests are flakes that #6907 fixes

**Review by tiina303 — CHANGES_REQUESTED** [review]
Test failures seem legit.

File: plugin-server/src/utils/db/db.ts [hunk 1]
```diff
@@ -497,27 +497,41 @@ export class DB {
...
-        const values = [...Object.values(unparsePersonPartial(update)), person.id]
+        const updateValues = Object.values(unparsePersonPartial(update))
 
-        const queryString = `UPDATE posthog_person SET ${Object.keys(update).map(
-            (field, index) => `"${sanitizeSqlIdentifier(field)}" = $${index + 1}`
-        )} WHERE id = $${Object.values(update).length + 1}`
+        // short circuit if there are no updates to be made
+        if (updateValues.length === 0) {
+            return client ? [] : person
+        }
```
Comment (tiina303) [inline plugin-server/src/utils/db/db.ts:505]: We have this check in the process-event side already & I'm not sure why, but it's bypassed when we're using kafka producer i.e. Clickhouse DB.
We always pass the full person's properties to this function from `updatePersonProperties` function at least.

**Comment by tiina303** [comment]
https://github.com/PostHog/posthog/pull/7007 <= you can merge this into here if that looks good for fixing tests + making us also not send ClickHouse update if nothing changed about props

**Comment by tiina303** [comment]
Please rebase on master first as I landed the props being added during person creation, which changes version values in tests.

**Review by tiina303 — APPROVED** [review]
@yakkomajuri I updated it based on merge needs + tests would love to get this in before the release leaving to you for final review

**Pull request merged**